quadcopter_brain/src/quadcopter_brain/quadcopter_brain.py

```diff
@@ -38,17 +41,88 @@ def go_to_waypoints(self, waypoint_data, time_to_sleep=15):
     def land(self):
         self.quadcopter.land()
 
-    def rc_land_on_fiducial(self):
-        dz = self.landing_site.center.position.z
-        while dz > 1:
-            dx = self.landing_site.center.position.x
-            dy = self.landing_site.center.position.y
-            rc_proportionally_navigate(dx, dy, dz)
-
-        self.land()
+    def send_rc_command(self, x_diff, y_diff, z_diff):
+        #Diff must be a value between 0 and 1
+        command = rc_command.RCCommand({'roll': x_diff,
+                                        'pitch': y_diff,
+                                        'throttle': z_diff})
+        self.quadcopter.send_rc_command(command)
+
+    def rc_square_dance(self):
+        rospy.loginfo("forward")
+        rc_command = rc_command.RCCommand({'pitch': 0.9})
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+        rospy.loginfo("still")
+        rc_command = rc_command.RCCommand()
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+
+        rospy.loginfo("right")
+        rc_command = rc_command.RCCommand({'roll': 0.9})
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+        rospy.loginfo("still")
+        rc_command = rc_command.RCCommand()
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+
+        rospy.loginfo("backward")
+        rc_command = rc_command.RCCommand({'pitch': 0.1})
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+        rospy.loginfo("still")
+        rc_command = rc_command.RCCommand()
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+
+        rospy.loginfo("left")
+        rc_command = rc_command.RCCommand({'roll': 0.1})
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
+        rospy.loginfo("still")
+        rc_command = rc_command.RCCommand()
+        self.quadcopter.send_rc_command(rc_command)
+        time.sleep(2)
 
-    def rc_proportionally_navigate(dx, dy, dz):
-        pass
+    def rc_land_on_fiducial(self):
+        found, _, _ = self.find_landing_site()
+        if found:
+            dz = self.landing_site.center.position.z
+            print dz
+            while dz > 1 and not rospy.is_shutdown():
+                dz = self.landing_site.center.position.z
+                dx = self.landing_site.center.position.x
+                dy = self.landing_site.center.position.y
+
+                self.proportional_position(dx, dy, dz)
+
+            self.send_rc_command(0.5, 0.5, 0.25)
+            # Currently enters RTL mode after script ends
+            # Wait 20 seconds to ensure landing
+            # TODO: change to a smarter land check
+            time.sleep(20)
+
+    def get_planar_speed(self, pos):
```

Comment:
From looking at this I don't know what it is - can we rename the function/variables to be more clear? And maybe include a docstring?